ffn/utils/proofreading.py

```diff
@@ -182,58 +260,293 @@ def __init__(self, objects, bad, num_to_prefetch=10, locations=None):
+    self.temp_coord_list = []
+
+  def set_keybindings(self) -> None:
+    """Set key bindings for the viewer."""
+    super().set_keybindings()
+    self.viewer.actions.add(
+        "merge0", lambda s: self.store_error_location(s, idx=0, mode="merger")
+    )
+    self.viewer.actions.add(
+        "merge1", lambda s: self.store_error_location(s, idx=1, mode="merger")
+    )
+    self.viewer.actions.add(
+        "split0", lambda s: self.store_error_location(s, idx=0, mode="split")
+    )
+    self.viewer.actions.add(
+        "split1", lambda s: self.store_error_location(s, idx=1, mode="split")
+    )
+    self.viewer.actions.add(
+        "delete_from_annotation", self.delete_location_from_annotation
+    )
+    self.viewer.actions.add(
+        "delete_last_entry", lambda s: self.delete_last_location()
+    )
+
+    with self.viewer.config_state.txn() as s:
+      s.input_event_bindings.viewer["keyw"] = "merge0"
+      s.input_event_bindings.viewer["shift+keyw"] = "merge1"
+      s.input_event_bindings.viewer["keyd"] = "split0"
+      s.input_event_bindings.viewer["shift+keyd"] = "split1"
+      s.input_event_bindings.viewer["control+keyv"] = "delete_from_annotation"
+      s.input_event_bindings.viewer["control+keyz"] = "delete_last_entry"
+
+  def get_id(self, mode: str) -> str:
+    """Generate a unique identifier for an error based on its type.
+
+    Args:
+      mode: Error type, either 'merge' or 'split'.
+
+    Returns:
+      A unique identifier string.
+    """
+    id_ = mode[0]
+    if any(self.seg_error_coordinates):
+      counter = int(max([x[1:] for x in self.seg_error_coordinates.keys()])) + 1
+    else:
+      counter = 0
+    id_ = id_ + str(counter)
+    return id_
+
+  def store_error_location(
+      self,
+      action_state: neuroglancer.viewer_config_state.ActionState,
+      mode: str,
+      idx: int = 0,
+  ) -> None:
+    """Store error locations.
+
+    Args:
+      action_state: State of the viewer during the action.
+      mode: Type of the error ('merger' or 'split').
+      idx: Indicates if it's the first or second coordinate (0 or 1).
+    """
+    location = self.get_cursor_position(action_state)
+    if location is None:
+      return
+
+    if idx == 1 and not self.temp_coord_list:
+      self.update_msg("You have not entered a first coord yet")
+      return
+
+    if idx == 0 and self.temp_coord_list:
+      self.temp_coord_list = []
+
+    self.temp_coord_list.append(location)
+
+    if idx == 1:
+      if self.temp_coord_list[0] == self.temp_coord_list[1]:
+        self.update_msg("You entered the same coordinate twice. Try again!")
+        self.temp_coord_list = []
+        return
+
+      identifier = self.get_id(mode=mode)
+      self.seg_error_coordinates.update({identifier: self.temp_coord_list})
+      self.annotate_error_locations(self.temp_coord_list, identifier)
+      self.temp_coord_list = []
+
+  def annotate_error_locations(
+      self, coordinate_lst: list[list[int]], id_: str
```

Comment:
Best not to include type names in the variable names (that's why we have type annotations). Maybe just coordinates instead of coordinate_lst?